src/main/java/org/traccar/handler/events/FuelEventHandler.java

```diff
@@ -55,13 +58,23 @@ public void onPosition(Position position, Callback callback) {
                     double threshold = AttributeUtil.lookup(
                             cacheManager, Keys.EVENT_FUEL_INCREASE_THRESHOLD, position.getDeviceId());
                     if (threshold > 0 && change >= threshold) {
-                        callback.eventDetected(new Event(Event.TYPE_DEVICE_FUEL_INCREASE, position));
+                        Map<String, Object> attributes = new HashMap<>();
+                        attributes.put("before", before);
+                        attributes.put("after", after);
+                        Event event = new Event(Event.TYPE_DEVICE_FUEL_INCREASE, position);
+                        event.setAttributes(attributes);
+                        callback.eventDetected(event);
```

Comment:
Instead of providing attributes as a map, you should use `set` function on the even object.